dotCMS/src/main/java/com/dotcms/storage/ContentletMetadataAPIImpl.java

```diff
@@ -0,0 +1,255 @@
+import com.liferay.util.StringPool;
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import io.vavr.control.Try;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Default implementation
+ * @author jsanca
+ */
+public class ContentletMetadataAPIImpl implements ContentletMetadataAPI {
+
+    private final FileStorageAPI fileStorageAPI;
+
+    public ContentletMetadataAPIImpl() {
+        this(APILocator.getFileStorageAPI());
+    }
+
+    @VisibleForTesting
+    public ContentletMetadataAPIImpl(final FileStorageAPI fileStorageAPI) {
+        this.fileStorageAPI = fileStorageAPI;
+    }
+
+    @Override
+    public ContentletMetadata generateContentletMetadata(final Contentlet contentlet,
+                                                         final Set<String> basicBinaryFieldNameSet,
+                                                         final Set<String> fullBinaryFieldNameSet) throws IOException {
+
+        final ImmutableMap.Builder<String, Map<String, Object>> fullMetadataMap  = new ImmutableMap.Builder();
+        final ImmutableMap.Builder<String, Map<String, Object>> basicMetadataMap = new ImmutableMap.Builder();
+        final  Map<String, com.dotcms.contenttype.model.field.Field>  fieldMap   = contentlet.getContentType().fieldMap();
+        /*
+		Verify if it is enabled the option to always regenerate metadata files on reindex,
+		enabling this could affect greatly the performance of a reindex process.
+		 */
+        final boolean alwaysRegenerateMetadata = Config
+                .getBooleanProperty("always.regenerate.metadata.on.reindex", false);
+        Map<String, Object> metadataMap        = null;
+
+        Logger.debug(this, ()-> "Generating the metadata for contentlet, id = " + contentlet.getIdentifier());
+
+        this.generateFullMetadata (contentlet, fullBinaryFieldNameSet, fullMetadataMap, fieldMap, alwaysRegenerateMetadata);
+        this.generateBasicMetadata(contentlet, basicBinaryFieldNameSet, fullBinaryFieldNameSet, basicMetadataMap, fieldMap, alwaysRegenerateMetadata);
+
+        return new ContentletMetadata(fullMetadataMap.build(), basicMetadataMap.build());
+    }
+
+    private void generateBasicMetadata(final Contentlet contentlet,
+                                       final Set<String> basicBinaryFieldNameSet,
+                                       final Set<String> fullBinaryFieldNameSet,
+                                       final ImmutableMap.Builder<String, Map<String, Object>> basicMetadataMap,
+                                       final Map<String, Field> fieldMap,
+                                       final boolean alwaysRegenerateMetadata) throws IOException {
+
+        final String storageType        = Config.getStringProperty("DEFAULT_STORAGE_TYPE", StorageType.FILE_SYSTEM.name());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "DEFAULT_STORAGE_TYPE" appears 4 times in this file; the first occurrence is on line 80](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)